code/go/0chain.net/miner/protocol_block.go

```diff
@@ -640,6 +652,12 @@ func txnIterHandlerFunc(mc *Chain,
 			logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", qe))
 			return true
 		}
+		cost := mc.EstimateTransactionCost(ctx, mc.GetLatestFinalizedBlock(), mc.GetLatestFinalizedBlock().ClientState, txn)
```

Comment:
If there's an Unmarshal error happens in the EstimateTransactionCost, the cost will be > MaxBlockCost and cause unexpected behaviour, the EstimateTransactionCost should return error.

Reply:
yeah, this is correct, overflow is possible

Reply:
fixed